src/dune_pkg/package_variable.mli

```diff
@@ -1,7 +1,12 @@
 open Import
 
 module Name : sig
-  type t
+  type t = OpamVariable.t
```

Comment:
Does this need to be opam-specific?

Reply:
My preference would be to internally represent `Name.t` with an `OpamVariable.t` to maintain compatibility with opam but not expose this fact in the api and make the conversion to opam explicit. This is the convention that most of our pkg code follows and it helps clarify which parts of pkg are opam-specific and which parts are dune's own pkg policy.